src/Core/Framework/Extensions/ExtensionDispatcher.php

```diff
@@ -38,7 +43,15 @@ public function publish(string $name, Extension $extension, callable $function):
         $this->dispatcher->dispatch($extension, self::pre($name));
 
         if (!$extension->isPropagationStopped()) {
-            $extension->result = $function(...$extension->getParams());
+            try {
+                $extension->result = $function(...$extension->getParams());
+            } catch (\Throwable $e) {
+                $extension->exception = $e;
+
+                $this->dispatcher->dispatch($extension, self::error($name));
+
+                throw $e;
+            }
```

Comment:
Not really happy with this, because:
- it modifies the stack trace as the Exception is re-thrown
- it does not allow yet for extensions to catch the Exception 🤔

Reply:
1. The stack trace is already messed up with the extension pattern right? So I would not bother about that too much, BTW honestly I'm not even sure about the stacktrace of a catcher and rethrown exception, is it created when the exception is thrown or created?
2. one rather straightforward way would be to allow to reset the exception to null from your extension, indicating that you handled it gracefully, and then not rethrow the exception? Of course we could also add a separate flag, that would allow multiple extensions to react to it